code/physics/physics.cpp

```diff
@@ -725,26 +725,34 @@ void physics_read_flying_controls( matrix * orient, physics_info * pi, control_i
 }
 
 
+#define WHACK_LIMIT 0.001f
+#define ROTVEL_WHACK_CONST 0.12f
+
+//	-----------------------------------------------------------------------------------------------------------
+// Returns true if this impulse is below the limit and should be ignored.
+bool whack_below_limit(const vec3d* impulse)
+{
+	return (fl_abs(impulse->xyz.x) < WHACK_LIMIT) && (fl_abs(impulse->xyz.y) < WHACK_LIMIT) &&
+		   (fl_abs(impulse->xyz.z) < WHACK_LIMIT);
+}
+
 // ----------------------------------------------------------------------------
 // physics_apply_whack applies an instaneous whack on an object changing
 // both the objects velocity and the rotational velocity based on the impulse
 // being applied.
 //
 //	input:	impulse		=>		impulse vector ( force*time = impulse = change in momentum (mv) )
-//				pos			=>		vector from center of mass to location of where the force acts
+//				pos			=>		vector from center of mass to location of where the force acts (in world coords)
 //				pi				=>		pointer to phys_info struct of object getting whacked
 //				orient		=>		orientation matrix (needed to set rotational impulse in body coords)
 //				mass			=>		mass of the object (may be different from pi.mass if docked)
 //
-#define WHACK_LIMIT	0.001f
-#define ROTVEL_WHACK_CONST 0.12
-void physics_apply_whack(vec3d *impulse, vec3d *pos, physics_info *pi, matrix *orient, float mass)
+void physics_calculate_whack(vec3d *impulse, vec3d *pos, physics_info *pi, matrix *orient, float mass, matrix *inv_moi)
```

Comment:
Rename this to `physics_calculate_and_apply_whack`.  I thought the functions would be entirely separate, but `calculate` feeds into `apply`.